geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/coprocessor/GeoMesaCoprocessor.scala

```diff
@@ -60,18 +58,17 @@ class GeoMesaCoprocessor extends GeoMesaCoprocessorService with Coprocessor with
       request: GeoMesaProto.GeoMesaCoprocessorRequest,
       done: RpcCallback[GeoMesaProto.GeoMesaCoprocessorResponse]): Unit = {
 
-    val notifier = new CancelCallback()
-    /*_*/controller.notifyOnCancel(notifier)/*_*/
+    val results = ArrayBuffer.empty[Array[Byte]]
 
-    val response: GeoMesaCoprocessorResponse = try {
-      val results = ArrayBuffer.empty[Array[Byte]]
-
-      if (!notifier.canceled.get) {
+    try {
+      if (!controller.isCanceled) {
         val options = GeoMesaCoprocessor.deserializeOptions(request.getOptions.toByteArray)
+        val timeout = options.get(GeoMesaCoprocessor.TimeoutOpt).map(_.toLong + System.currentTimeMillis())
         val aggregator = {
           val classname = options(GeoMesaCoprocessor.AggregatorClass)
           Class.forName(classname).newInstance().asInstanceOf[HBaseAggregator[_]]
         }
+        logger.debug(s"Initializing aggregator $aggregator with options ${options.mkString(", ")}")
```

Comment:
Just as a sanity check, the timeout config is logged, right?

Reply:
yeah it would be in the options map